packages/demos/stories/TableColumnsDragDrop.tsx

```diff
@@ -0,0 +1,286 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayTable from '@clayui/table';
+import classNames from 'classnames';
+import React from 'react';
+import {
+	DropTargetMonitor,
+	XYCoord,
+	useDrag,
+	useDragLayer,
+	useDrop,
+} from 'react-dnd';
+
+const TABLE_COLUMN_DND_TYPE = 'tableColumn';
+
+const initialTableColumns = [
+	{
+		data: ['Red, white & blue', 'Tricolore', 'Stars & stripes'],
+		id: 'flag',
+		title: 'Flag',
+	},
+	{
+		data: ['Europe', 'Europe', 'North America'],
+		id: 'region',
+		title: 'Region',
+	},
+	{
+		data: ['Croatia', 'Italy', 'USA'],
+		id: 'country',
+		title: 'Country',
+	},
+	{
+		data: [
+			'Lorem ipsum dolor, sit amet consectetur adipisicing elit.',
+			'Lorem ipsum dolor, sit amet consectetur adipisicing elit.',
+			'Lorem ipsum dolor, sit amet consectetur adipisicing elit.',
+		],
+		id: 'description',
+		title: 'Description',
+	},
+];
+
+const dragLayerStyles: React.CSSProperties = {
+	height: '100%',
+	left: 0,
+	pointerEvents: 'none',
+	position: 'fixed',
+	top: 0,
+	width: '100%',
```

Comment:
Any way we can get the width of the row itself? It looks a little odd to have the drag layer with 100% width.